dbos/_dbos.py

```diff
@@ -945,16 +945,45 @@ def fn() -> None:
         return cls.retrieve_workflow(workflow_id)
 
     @classmethod
-    def restart_workflow(cls, workflow_id: str) -> WorkflowHandle[Any]:
+    def get_max_function_id(cls, workflow_uuid: str) -> Optional[int]:
```

Comment:
We shouldn't add this to the public API (which is what the class methods are)&mdash just put this logic inside `fork_workflow`. To shorten it, we can make the `get_max_function_id` return 0 instead of None if nothing is found.